[Fix]: #4 [Fix]: #5

rhl-bthr opened the pull request:
**Fixed in this PR:** #4  and #5(partial)

**Changes made:**
 - [X] Remove multiple instances of stop button
 - [X] Replace `Gtk.Entry` in toolbar with `ActivityToolBarButton`
 - [X] Place `StopButton` at the rightmost end, keeping in mind the general trend
 - [X] Hiding Notebook on fullscreen mode (Ref: [here](https://github.com/sugarlabs/terminal-activity/issues/25))

@cristian99garcia, Kindly review

Commit:
Removed multiple instances of stop button

 - Also, using activity.widgets.StopButton

Commit:
Add activityToolbarButton

 - Replace Gtk.Entry instance with activityToolBar

Commit:
Hide notebook tabs on fullscreen

 - Hide notebook tabs on fullscreen
 - Show tabs on exiting fullscreen

quozl commented:
Reviewed.

Commit:
Fix: use of GTK3 instead of gtk2

 - Possibly left while porting

Commit:
Removed debug toolbar

Reason:
 - Since it contains only the debug button which is present on the main toolbar
 - Can be reintroduced if more buttons are added

Commit:
change position of Notebook from bottom to top

Commit:
Give names to tabs. Don't close if only one tab is present

 - Using the close tab button will not have any effect if only one tab is present
 - Tabs are named in the followinf form:
     Tab <num>
 - Tab labels get changed each time a tab is closed

Pro-Panda commented:
**Updates in PR:**
 - [X] Fix bug (possibly left during GTK3 port)
 - [X] Remove debug toolbar since it was redundant (debug button is present on the main toolbar)
 - [X] Change Notebook position from bottom to top
 - [X] Label tabs in the form `Tab <num>`. Tabs names get auto-adjusted if any tab is closed
 - [X] Attempting to close the last remaining tab does not do anything
        Only the stop button should be closing the activity
        Idea borrowed from activities like cedit and terminal

Kindly re-review

quozl commented:
Reviewed.  Not tested.

Commit:
Fix: Activity Title not being set

 - Also Remove redundant function and its call

quozl commented:
Reviewed.  Not tested.  @cristian99garcia, do you have any review comments or will you merge?

Commit:
Fix: Issues in building pot files

 - Add license in activity .info
 - Add License

Note: setup.py still throwing exceptions

quozl commented:
Please use COPYING rather than LICENSE.  COPYING is the most frequently used in Sugar activities.

Commit:
Set License to COPYING

Pro-Panda commented:
Updated.
I fixed all issues that came while building `pot` file, in 822a284
But trying to build an xo(`python setup.py dist_xo`) returns fatal errors which I couldn't figure how to approach.

Logs: https://paste.fedoraproject.org/paste/f8AGB96S~GwhvOz45LAnOg

Kindly review logs and guide me on how to approach this

quozl commented:
Message shows duplicate strings in [`zh_CN.po`](https://github.com/sugarlabs/pydebug-activity/blob/master/po/zh_CN.po) which are likely cause of error.

`vi.po` is from Terminal activity.

`es.po` looks reasonable.

My guess is that the translations were copied from Terminal and are not maintained.

PyDebug is not listed at translate.sugarlabs.org.

You may find `gist` package useful.

Pro-Panda commented:
Sorry, but I couldn't follow.
What is the `gist` package ?

quozl commented:
https://packages.ubuntu.com/search?keywords=gist

Pro-Panda commented:
Oh, I will use that for uploading logs from now on. Thanks 
Regarding the issue, I was able to follow what you explained, but couldn't follow what needs to be done to fix it

quozl commented:
I'm not sure what to do about it either, but at least you know how it is caused.   You might remove the duplicate messages with a text editor or by running `python setup.py genpot`.  Changes from the `genpot` step would normally be committed as "Regenerate POT File" for translators to pick up, but as the project is not enabled for translation at translate.sugarlabs.org I'm not sure why you'd bother.

Commit:
Fix: Remove duplicate strings

 - Raised errors while trying to build the xo

cristian99garcia commented:
Hello @Pro-Panda, sorry for the late answer, and sorry but currently I can't test it :( I haven't installed sugar on my pc and I have no time to do that.

@quozl do you think that this PR is ready to be merged?

quozl commented:
@cristian99garcia, in my opinion no need for reviewer to test, see new [Guide for Reviewers](https://github.com/sugarlabs/sugar-docs/blob/master/src/contributing.md#guide-for-reviewers).  Test can be left to activity maintainer during release, see new [Modifying Activities](https://github.com/sugarlabs/sugar-docs/blob/master/src/contributing.md#modifying-activities).

Yes, I've reviewed already and think merge is due.  I shall do that now.

cristian99garcia commented:
Thanks @quozl, @Pro-Panda
